node/directives/validateAdminUserAccess.ts

```diff
@@ -0,0 +1,136 @@
+    const { resolve = defaultFieldResolver } = field
+
+    field.resolve = async (
+      root: any,
+      args: any,
+      context: Context,
+      info: any
+    ) => {
+      const {
+        vtex: { adminUserAuthToken, logger },
+      } = context
+
+      // get metrics data
+      const operation = field?.astNode?.name?.value ?? context?.request?.url
+      const userAgent = context?.request?.headers['user-agent'] as string
+      const caller = context?.request?.headers['x-vtex-caller'] as string
+      const forwardedHost = context?.request?.headers[
+        'x-forwarded-host'
+      ] as string
+
+      // set metric fields with initial data
+      let metricFields: AuthAuditMetric = {
+        operation,
+        forwardedHost,
+        caller,
+        userAgent,
+      }
+
+      const { hasAdminToken, hasValidAdminToken, hasValidAdminTokenFromStore } =
+        await validateAdminToken(context, adminUserAuthToken as string)
+
+      // add admin token metrics
+      metricFields = {
+        ...metricFields,
+        hasAdminToken,
+        hasValidAdminToken,
+        hasValidAdminTokenFromStore,
+      }
+
+      // allow access if has valid admin token
+      if (hasValidAdminToken) {
+        sendAuthMetric(
+          logger,
+          new AuthMetric(
+            context?.vtex?.account,
+            metricFields,
+            'ValidateAdminUserAccessAudit'
+          )
+        )
+
+        return resolve(root, args, context, info)
+      }
+
+      // If there's no valid admin token on context, search for it on header
+      const { hasAdminTokenOnHeader, hasValidAdminTokenOnHeader } =
+        await validateAdminTokenOnHeader(context)
+
+      // add admin header token metrics
+      metricFields = {
+        ...metricFields,
+        hasAdminTokenOnHeader,
+        hasValidAdminTokenOnHeader,
+      }
+
+      // allow access if has valid admin token
+      if (hasValidAdminTokenOnHeader) {
+        sendAuthMetric(
+          logger,
+          new AuthMetric(
+            context?.vtex?.account,
+            metricFields,
+            'ValidateAdminUserAccessAudit'
+          )
+        )
+
+        return resolve(root, args, context, info)
+      }
+
+      const { hasApiToken, hasValidApiToken, hasValidApiTokenFromStore } =
+        await validateApiToken(context)
+
+      // add API token metrics
+      metricFields = {
+        ...metricFields,
+        hasApiToken,
+        hasValidApiToken,
+        hasValidApiTokenFromStore,
+      }
+
+      // allow access if has valid API token
+      if (hasValidApiToken) {
+        sendAuthMetric(
+          logger,
+          new AuthMetric(
+            context?.vtex?.account,
+            metricFields,
+            'ValidateAdminUserAccessAudit'
+          )
+        )
+
+        return resolve(root, args, context, info)
+      }
+
+      // deny access if no tokens were provided
+      if (!hasAdminToken && !hasApiToken) {
```

Comment:
Missing the `hasAdminTokenOnHeader` here.

Reply:
Fixed